### PR TITLE
Add example on fill(anArray,n) in array.jl

### PR DESCRIPTION
**Pull request by PaulSoderlind**

### CODE REVIEW

File: base/array.jl [hunk 1]
```diff
@@ -387,8 +387,22 @@ julia> fill(1.0, (5,5))
...
+julia> X = fill(zeros(2),2)
+2-element Array{Array{Float64,1},1}:
+ [0.0, 0.0]
+ [0.0, 0.0]
+julia> X[1][1] = 99;
+julia> X
+2-element Array{Array{Float64,1},1}:
+ [99.0, 0.0]
+ [99.0, 0.0]
+```
```
Comment (StefanKarpinski) [inline base/array.jl]:
```suggestion
The single `zeros(2)` object which is stored at both positions in `X` is mutated.
```

File: base/array.jl [hunk 2]
```diff
@@ -387,8 +387,22 @@ julia> fill(1.0, (5,5))
...
+
+`fill(Foo(),dims)` will return an array filled with the result of evaluating `Foo()` once.
```
Comment (StefanKarpinski) [inline base/array.jl]:
```suggestion
Similarly, `fill(Foo(),dims)` will return an array filled with the result of evaluating `Foo()` once.
If you want an array with independent `Foo` objects, you can use a comprehension:
`[Foo() for _ in dims]`.
```

**Comment by StefanKarpinski** [comment]
Thanks for the contribution to improve the docs! Much appreciated 👍

File: base/array.jl [hunk 3]
```diff
@@ -387,8 +387,22 @@ julia> fill(1.0, (5,5))
  1.0  1.0  1.0  1.0  1.0
 ```
 
-If `x` is an object reference, all elements will refer to the same object. `fill(Foo(),
-dims)` will return an array filled with the result of evaluating `Foo()` once.
+If `x` is an object reference, all elements will refer to the same object.
```
Comment (KristofferC) [inline base/array.jl]: I don't really understand the original comment here. What is an `object reference`? Feels like this should just be

```
!!! note
   All elements in the resulting array will refer to the same object.
```

**Comment by ViralBShah** [comment]
I'd like to get this merged, but there are some nice suggestions, that I can't easily commit. Should we merge this one and then apply the suggestions here in a separate PR?

**Review by PaulSoderlind — COMMENTED** [review]
Good changes.

File: base/array.jl [hunk 4]
```diff
@@ -387,8 +387,22 @@ julia> fill(1.0, (5,5))
...
+
+# Examples
+```jldoctest
+julia> X = fill(zeros(2),2)
+2-element Array{Array{Float64,1},1}:
+ [0.0, 0.0]
+ [0.0, 0.0]
+julia> X[1][1] = 99;
```
Comment (rfourquet) [inline base/array.jl:398]: This should have a newline inserted above and below.

Comment (PaulSoderlind) [reply base/array.jl:398]: Sorry for asking (I really don't understand the PR machinery works): do I need to do anything to incorporate your comment?

Comment (rfourquet) [reply base/array.jl:398]: Usually it's possible for reviewer to make a concrete code change suggestion, which can be easily committed, but I didn't manage this time. So yeah, you would need to update the PR but I'm not really sure how, as I see that it's made from an "unknown repository"...

Comment (PaulSoderlind) [reply base/array.jl:398]: @ViralBShah Should I manually combine these changes and make a new PR? Or, is there another way?  /Paul

Comment (ViralBShah) [reply base/array.jl:398]: I think making a new PR is the easiest way.

**Comment by fredrikekre** [comment]
Rebased #35683

**Pull request closed without merging**